src/ImageSharp.Web/Processors/WebProcessingExtensions.cs

```diff
@@ -33,15 +35,29 @@ internal static class WebProcessingExtensions
             CommandParser commandParser,
             CultureInfo culture)
         {
-            if (commands.Count != 0)
+            var commandKeys = new List<string>(commands.Keys);
+
+            foreach (IImageWebProcessor processor in processors.GetBySupportedCommands(commandKeys))
             {
-                foreach (IImageWebProcessor processor in processors)
-                {
-                    source = processor.Process(source, logger, commands, commandParser, culture);
-                }
+                source = processor.Process(source, logger, commands, commandParser, culture);
             }
 
             return source;
         }
+
+        /// <summary>
+        /// Sorts the processors according to the first supported command and removes processors without supported commands.
+        /// </summary>
+        /// <param name="processors">The collection of available processors.</param>
+        /// <param name="commands">The parsed collection of processing commands.</param>
+        /// <returns>
+        /// The sorted proccessors that supports any of the specified commands.
+        /// </returns>
+        public static IEnumerable<IImageWebProcessor> GetBySupportedCommands(this IEnumerable<IImageWebProcessor> processors, List<string> commands)
+            => processors
+                .GroupBy(p => commands.FindIndex(c => p.Commands.Contains(c, StringComparer.OrdinalIgnoreCase))) // Get index of first supported command
```

Comment:
I've greatly simplified this check (compared to the first commits) and although `FindIndex()` is an O(n) operation, unknown commands are already stripped and this will only be run when the image needs to get processed (i.e. is not already cached).

Reply:
Can't have Linq on a hot path like this.

Reply:
I'll rewrite this into a more verbose, non-LINQ, version 👍🏻